scene/main/canvas_item.cpp

```diff
@@ -72,6 +72,15 @@ bool CanvasItem::is_visible_in_tree() const {
 		p = p->get_parent_item();
 	}
 
+	const Node *n = get_parent();
```

Comment:
I think this may be getting to be a bit expensive over time. Since we are sending the visibility changed signal and notifications, maybe at some point this value should just be cached.